src/sites/twitch-twilight/modules/host_button.js

```diff
@@ -0,0 +1,336 @@
+'use strict';
+
+// ============================================================================
+// Host Button
+// ============================================================================
+
+import Module from 'utilities/module';
+import {createElement as e} from 'utilities/dom';
+
+export default class HostButton extends Module {
+	constructor(...args) {
+		super(...args);
+
+		this.should_enable = true;
+
+		this.inject('site.fine');
+		this.inject('metadata');
+
+		this.metadata.definitions.host = {
+			order: 150,
+			button: true,
+
+			disabled: () => {
+				return this._host_updating;	
+			},
+			
+			click: (data, event) => {
+				if (data.channel) this.sendHostUnhostCommand(data.channel.login);
+			},
+
+			popup: async (data, tip) => {
+				const vue = this.resolve('vue'),
+					_host_options_vue = import(/* webpackChunkName: "host-options" */ './host-options.vue'),
+					_autoHosts = this.fetchAutoHosts(),
+					_autoHostSettings = this.fetchAutoHostSettings();
+				
+				const [, host_options_vue, autoHosts, autoHostSettings] = await Promise.all([vue.enable(), _host_options_vue, _autoHosts, _autoHostSettings]);
+
+				this._auto_host_tip = tip;
+				tip.element.classList.remove('pd-1');
+				tip.element.classList.add('tw-balloon--lg');
+				vue.component('host-options', host_options_vue.default);
+				return this.buildAutoHostMenu(vue, autoHosts, autoHostSettings, data.channel);
+			},
+
+			label: data => {
+				if (this._host_updating) {
+					return '<figure class="ffz-i-zreknarf loading"/>';
+				}
+
+				return (this._last_hosted_channel && this.isChannelHosted(data.channel && data.channel.login)) ? 'Unhost' : 'Host';
+			},
+
+			tooltip: () => {
+				return `Currently hosting: ${this._last_hosted_channel || 'None'}`;
+			}
+		};
+	}
+
+	sendHostUnhostCommand(channel) {
+		if (!this._chat_con) return;
```

Comment:
Should there be feedback if we fail to grab the chat connection for some reason? Users might get frustrated if they click and nothing seems to happen.